src/overlays/actors/ovl_En_Ma_Yto/z_en_ma_yto.c

```diff
@@ -21,126 +83,1425 @@ const ActorInit En_Ma_Yto_InitVars = {
+}
+
+void EnMaYto_SetRomaniFaceExpression(EnMaYto* this, s16 overrideEyeTexIndex, s16 mouthTexIndex) {
+    EnMaYts* romani = (EnMaYts*)this->actor.child;
+
+    if ((romani != NULL) && (romani->actor.id == ACTOR_EN_MA_YTS)) {
+        romani->overrideEyeTexIndex = overrideEyeTexIndex;
+        romani->mouthTexIndex = mouthTexIndex;
+    }
+}
+
+void EnMaYto_SetFaceExpression(EnMaYto* this, s16 overrideEyeTexIndex, s16 mouthIndex) {
+    this->overrideEyeTexIndex = overrideEyeTexIndex;
+    this->mouthTexIndex = mouthIndex;
+}
+
+void EnMaYto_InitFaceExpression(EnMaYto* this) {
+    if (CURRENT_DAY == 1 || (gSaveContext.weekEventReg[0x16] & 1)) {
+        EnMaYto_SetFaceExpression(this, 0, 1);
+        EnMaYto_SetRomaniFaceExpression(this, 0, 0);
+    } else {
+        EnMaYto_SetFaceExpression(this, 5, 2);
+        EnMaYto_SetRomaniFaceExpression(this, 1, 2);
+    }
+}
+
+s32 EnMaYto_HasSpokeToPlayerToday(void) {
+    switch (CURRENT_DAY) {
+        case 1:
+            if (gSaveContext.weekEventReg[0xD] & 0x4) {
+                return true;
+            }
+            break;
+
+        case 2:
+            if (gSaveContext.weekEventReg[0xD] & 0x8) {
+                return true;
+            }
+            break;
+
+        case 3:
+            if (gSaveContext.weekEventReg[0xD] & 0x10) {
+                return true;
+            }
+            break;
+    }
+    return false;
+}
+
+s32 EnMaYto_HasSpokeToPlayer(void) {
+    // Please note each case doesn't have their respective `break`s.
+    switch (CURRENT_DAY) {
+        case 3:
+            if (gSaveContext.weekEventReg[0xD] & 0x10) {
+                return true;
+            }
+
+        case 2:
+            if (gSaveContext.weekEventReg[0xD] & 0x8) {
+                return true;
+            }
+
+        case 1:
+            if (gSaveContext.weekEventReg[0xD] & 0x4) {
+                return true;
+            }
+    }
+    return false;
+}
+
+void EnMaYto_SetTalkedFlag(void) {
+    switch (CURRENT_DAY) {
+        case 1:
+            gSaveContext.weekEventReg[0xD] |= 0x4;
+            break;
+
+        case 2:
+            gSaveContext.weekEventReg[0xD] |= 0x8;
+            break;
+
+        case 3:
+            gSaveContext.weekEventReg[0xD] |= 0x10;
+            break;
+    }
+}
+
+void EnMaYto_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnMaYto* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    EnMaYto_UpdateCollision(this, globalCtx);
+    func_80B90C78(this, globalCtx);
+}
+
+s32 EnMaYto_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                             Actor* thisx) {
+    EnMaYto* this = THIS;
+    Vec3s sp4;
+
+    if (limbIndex == MA2_LIMB_HEAD) {
+        sp4 = this->unk_1D8.unk_08;
+
+        rot->x += sp4.y;
+        rot->z += sp4.x;
+    } else if (limbIndex == MA2_LIMB_TORSO) {
+        if (this->skelAnime.animCurrentSeg != &D_06007E28 && this->skelAnime.animCurrentSeg != &D_06003D54) {
```

Comment:
This if could be put inside the else-if, unless you specifically wanted to keep the outer nest to limbIndex

Reply:
Yeah, I think it is better to leave the outer `if-else` chain to check limb indexes.